scripts/build_openssl.bat

```diff
@@ -1,74 +1,93 @@
 ::
 :: Build OpenSSL
 ::
+:: Prerequisite:
+:: - VC 2015 or 2017
+:: Arguments:
+:: - x86 / x64
+:: - Debug / Release
+:: - vs14 / vs15
 
-@echo on
-set OPENSSL_DIR=openssl-1.1.1b
-set CURL_DIR=curl-7.58.0
+@echo off
+set OPENSSL_VERSION=1.1.1b
+call %*
+goto :EOF
+
+:get_version
+    set version=%openssl_version%
+    goto :EOF
+
+:build
+setlocal
+set OPENSSL_DIR=openssl-%OPENSSL_VERSION%
+
+if defined GITHUB_ACTIONS (
+    set PERL_EXE=C:\Strawberry\perl\bin\perl
+) else (
+    set PERL_EXE=perl
+)
+echo === PERL_EXE: %PERL_EXE%
 
 set platform=%1
 set build_type=%2
 set vs_version=%3
+set dynamic_runtime=%4
 
 set scriptdir=%~dp0
-call "%scriptdir%\_init.bat" %platform% %build_type% %vs_version%
+set "path=%scriptdir%..\ci\tools;%path%"
+
+call "%scriptdir%_init.bat" %platform% %build_type% %vs_version%
 if %ERRORLEVEL% NEQ 0 goto :error
 set curdir=%cd%
 
 if "%platform%"=="x64" (
-	set openssl_target=VC-WIN64A
-	set engine_dir=Program Files
+    set openssl_target=VC-WIN64A
+    set engine_dir=Program Files
 )
 if "%platform%"=="x86" (
-	set openssl_target=VC-WIN32
-	set engine_dir=Program Files (x86^)
+    set openssl_target=VC-WIN32
+    set engine_dir=Program Files (x86^)
 )
 
 set crypto_target_name=
 set ssl_target_name=
-set curl_target_name=
 if "%build_type%"=="Debug" (
-	set openssl_debug_option=--debug
+    set openssl_debug_option=--debug
 )
 if "%build_type%"=="Release" (
-	set openssl_debug_option=
+    set openssl_debug_option=
 )
 set crypto_target_name=libcrypto_a.lib
 set ssl_target_name=libssl_a.lib
-set curl_target_name=libcurl_a.lib
 
-call "%scriptdir%\utils.bat" :setup_visual_studio %vs_version%
+call "%scriptdir%utils.bat" :setup_visual_studio %vs_version%
 
 echo === building openssl: %curdir%\..\deps\%OPENSSL_DIR%
-cd "%scriptdir%\..\deps
-tar -x -z -f openssl-1.1.1b.tar.gz
-cd "%scriptdir%\..\deps\%OPENSSL_DIR%"
-perl Configure %openssl_debug_option% %openssl_target% no-shared
+cd "%scriptdir%..\deps
+rd /s /q %OPENSSL_DIR%
+7z x openssl-1.1.1b.zip
```

Comment:
@rvishureddy , I've added the zip file because tar cannot be extracted by the old Windows machine.